File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Set<Integer> window = new HashSet<>();
        int left = 0;

        for (int right = 0; right < nums.length; right++) {
            if (window.contains(nums[right])) {
                return true;
            }

            window.add(nums[right]);

            while (right - left >= k) {
                window.remove(nums[left]);
                left++;
            }
        }

        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.kt
class Solution {
    fun containsNearbyDuplicate(nums: IntArray, k: Int): Boolean {
        val window = mutableSetOf<Int>()
        var left = 0

        for (right in nums.indices) {
            if (nums[right] in window) return true
            
            window.add(nums[right])

            if (right - left >= k) {
                window.remove(nums[left])
                left++
            }
        }

        return false
    }
}